packages/ag-charts-community/src/chart/dom/domManager.ts

```diff
@@ -241,10 +255,51 @@ export class DOMManager extends BaseManager<Events['type'], Events> {
         });
     }
 
+    /** Get the main chart area client bound rect. */
     getBoundingClientRect() {
         return this.rootElements['canvas'].element.getBoundingClientRect();
     }
 
+    /**
+     * Get the client bounding rect for overlay elements that might float outside the bounds of the
+     * main chart area.
+     */
+    getOverlayClientRect() {
+        const window = getWindow();
+        const windowBBox = new BBox(0, 0, window.innerWidth, window.innerHeight);
+        const container = this.getRawOverlayClientRect();
+
+        if (container == null) return windowBBox.toDOMRect();
+
+        const containerBBox = BBox.fromDOMRect(container);
+        return windowBBox.intersection(containerBBox)?.toDOMRect() ?? NULL_DOMRECT;
+    }
+
+    private getRawOverlayClientRect() {
+        let element: HTMLElement | null = this.element;
+
+        // Try and find a parent which will clip rendering of children - if found we should restrict
+        // to that elements bounding box.
+        while (element != null) {
+            const styleMap = element.computedStyleMap?.();
+            const overflowX = styleMap?.get('overflow-x')?.toString();
+            const overflowY = styleMap?.get('overflow-y')?.toString();
+
+            if (overflowX !== 'visible' || overflowY !== 'visible') {
+                return element.getBoundingClientRect();
+            }
+
+            element = element.parentElement;
+        }
+
+        // If in a shadow-DOM case, use the shadow-DOMs bounding-box, intersected with the window
+        // viewport.
```

Comment:
What is the expected behaviour for a shadow dom? Should the tooltip be able to be outside?

Reply:
Right now I would expect that the tooltip cannot escape the bounds of the shadow DOM.

Reply:
Does this check need to be moved before the while loop then?

Reply:
No, we still want to respect `overlow` on parent DOM nodes inside the shadow DOM - it's just if we reach the shadow-DOM root that we want to behave differently.